app/controllers/api/base_controller.rb

```diff
@@ -65,6 +67,28 @@ def response_with_mfa_warning(response)
     message
   end
 
+  def render_error_if_user_mfa_requirement_unmet(user)
```

Comment:
This method imo feels a bit long and complicated. It's doing many things (checking the mfa levels and possibly rendering it) and the output is different depending on if the requirement is unmet. Maybe we should think about refactoring this into multiple methods (eg. maybe have a method that renders mfa_required_not_yet_enabled error, and another renders mfa_required_weak_level_enabled? error).